src/reports/reporter.ts

```diff
@@ -11,25 +13,44 @@ export enum EventName {
   START_BINARY = "tabnine-binary-run",
 }
 
+let reporter: TelemetryReporter;
+
 export function initReporter(
   context: ExtensionContext,
   id: string,
   version: string,
   key: string
 ): void {
+  if (inTestMode) return;
+
   reporter = new TelemetryReporter(id, version, key);
   context.subscriptions.push(reporter);
 }
 
 export function report(event: EventName): void {
-  reporter.sendTelemetryEvent(event);
+  if (inTestMode) return;
+
+  void getReportData().then((data) => {
+    reporter.sendTelemetryEvent(event, data ?? {});
+  });
 }
 
 export function reportErrorEvent(event: EventName, error: Error): void {
-  reporter.sendTelemetryErrorEvent(event, { error: error.message });
+  if (inTestMode) return;
+
+  void getReportData().then((data) => {
```

Comment:
set 'data'  to default  `{}` value and then just use `{ ...data, error: error.message }`